cloud.foundry.cli/src/test/java/cloud/foundry/cli/operations/ApplicationsOperationsTest.java

```diff
@@ -223,6 +216,47 @@ public void testRemoveApplicationShouldThrowNullPointerExceptionWhenApplicationN
         assertThrows(NullPointerException.class, () -> applicationsOperations.remove(null));
     }
 
+    @Test
+    public void testRenameOnCurrentNameNullThrowsNullPointerException() {
+        //given
+        ApplicationsOperations applicationsOperations = new ApplicationsOperations(
+                mock(DefaultCloudFoundryOperations.class));
+
+        //when
+        assertThrows(NullPointerException.class, () ->
+                applicationsOperations.rename("appName", null));
+    }
+
+    @Test
+    public void testRenameOnNewNameNullThrowsNullPointerException() {
+        //given
+        ApplicationsOperations applicationsOperations = new ApplicationsOperations(
+                mock(DefaultCloudFoundryOperations.class));
+
+        //when
+        assertThrows(NullPointerException.class, () ->
+                applicationsOperations.rename("appName", null));
+    }
+
+    @Test
+    public void testRename() {
+        // given
+        DefaultCloudFoundryOperations cfoMock = mock(DefaultCloudFoundryOperations.class);
+        Applications applicationsMock = mock(Applications.class);
+        when(cfoMock.applications()).thenReturn(applicationsMock);
+
+        when(applicationsMock.rename(any())).thenReturn(Mono.empty());
+
+        ApplicationsOperations applicationsOperations = new ApplicationsOperations(cfoMock);
+
+        // when
+        Mono<Void> request = applicationsOperations.rename("newName", SOME_APPLICATION);
+
+        // then
+        assertThat(request, notNullValue());
+        verify(applicationsMock, times(1)).rename(any(RenameApplicationRequest.class));
```

Comment:
Is it possible to check whether a RenameApplicationRequest with correct values (old and new name) was passed?

Reply:
I don't think so, because the RenameApplicationRequest is built every time new and there is no easy way to access it.

Reply:
No problem, I managed to solve it :)